x-pack/legacy/plugins/apm/public/selectors/chartSelectors.ts

```diff
@@ -158,12 +159,18 @@ export function getTpmSeries(
   const bucketKeys = tpmBuckets.map(({ key }) => key);
   const getColor = getColorByKey(bucketKeys);
 
+  const { avg } = apmTimeseries.responseTimes;
+
+  if (!tpmBuckets.length && avg.length) {
+    return getEmptySeries(avg[0].x, avg[avg.length - 1].x);
+  }
```

Comment:
I'm not sure I understand this logic :)
As I read it an empty state series is returned if there are no tpm buckets but there is data for avg series. 
What happens when there is no data for either tpm buckets or avg series?

And how come `getEmptySeries` is necessary here, when it's not necessary for the other charts?

Reply:
I should add a comment here. The aggs in `apmTimeSeries` (`avg`,`p95`,`p99`) always return date buckets with null values, but `tpmBuckets` is from a term aggregation, meaning we get no date buckets, because we have no terms. We could send an empty series from the backend in those cases as well, maybe that is more obvious?

Reply:
The empty states still look off. The tick marks on the y-axis are missing (halfway and bottom)

<img width="2402" alt="Screenshot 2019-07-08 at 11 56 11" src="https://user-images.githubusercontent.com/209966/60801532-6fc6f200-a177-11e9-80a1-67b57ad4ceec.png">

Reply:
@sqren hmm, yeah, I hoped the `yTickValues` prop would fix this, but it's probably because there are only `null` coords. `getEmptySeries` creates a timeseries with coordinates of `y: 1`. We could fix it by reinstating `getEmptySeries` inside of CustomPlot again. Do you think it's worth it?

Reply:
> Do you think it's worth it?

Hmm.. I think so. There are (minor) regressions to the design and I'm seeing this error:

<img width="817" alt="Screenshot 2019-07-08 at 13 50 52" src="https://user-images.githubusercontent.com/209966/60808001-8a08cc00-a187-11e9-8835-269d2bd5ba78.png">

Reply:
@sqren yeah, the alternative would be to hide the gridlines and y axis in that case, like how it was in the initial implementation of this PR I think. I'll put back the empty series w/ a value of 1 (but it feels kinda gross).

Reply:
As discussed via Zoom, we decided to drop the YAxis and HorizontalGridLines if there's no data. The cost of the alternative is Too Damn High™, at least in terms of maintainability.